migen/build/tools.py

```diff
@@ -58,3 +59,32 @@ def subprocess_call_filtered(command, rules, *, max_matches=1, **kwargs):
                               rules, max_matches):
             sys.stdout.write(line)
     return proc.returncode
+
+
+if sys.platform == "cygwin":
+    cygwin1 = ctypes.CDLL("/usr/bin/cygwin1.dll")
+    cygwin_conv_path_proto = ctypes.CFUNCTYPE(
+        ctypes.c_ssize_t, # Return
+        ctypes.c_uint, # what
+        ctypes.c_void_p, # from
+        ctypes.c_void_p, # to
+        ctypes.c_size_t) # size
+    cygwin_conv_path = cygwin_conv_path_proto(("cygwin_conv_path", cygwin1),
+        ((1, "what"),
+        (1, "from"),
+        (1, "to"),
+        (1, "size")))
+
+
+    def cygpath_to_windows(path):
+        what = ctypes.c_uint(0) # CCP_POSIX_TO_WIN_A
+        fro = ctypes.c_char_p(path.encode('utf-8'))
+        to = ctypes.byref(ctypes.create_string_buffer(260))
+        size = ctypes.c_size_t(260)
+
+        cygwin_conv_path(what, fro, to, size)
+        return ctypes.cast(to, ctypes.c_char_p).value.decode('utf-8')
+
+
+def sanitize(p):
```

Comment:
Should be called ``insanitize``, or something less negative and more descriptive, e.g. ``cygpath``. Otherwise, acceptable.

Reply:
And you can simply define that function in the test here https://github.com/m-labs/migen/pull/88/files#diff-c5c08e937830e0e7a1cb6046624e3a81R64
...and add an ``else:`` branch that defines it to identity.